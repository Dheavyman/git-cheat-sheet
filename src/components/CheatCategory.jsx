import React from 'react';

import Cheat from './Cheat';

/**
 * Cheat category component
 *
 * @param {string} category - Cheat category
 * @param {string} cheats - Cheats in a category
 *
 * @returns {object} React element
 */
const CheatCategory = ({ category, cheats}) => {
  return (
    <div className="category-card">
      <h6 className="category">{category}</h6>
        {cheats.map(cheat => (
          <Cheat key={cheat._id} cheat={cheat} />
        ))}
    </div>
  );
}

export default CheatCategory;
