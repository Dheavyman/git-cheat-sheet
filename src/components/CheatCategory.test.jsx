import React from 'react';
import TestRenderer from 'react-test-renderer';

import CheatCategory from './CheatCategory';

describe('CheatCategory component', () => {
  const props = {
    category: 'cheat category',
    cheats: [{
      _id: 'cheat id',
      description: 'cheat description',
      command: 'cheat command',
    }, {
      _id: 'another cheat id',
      description: 'another cheat description',
      command: 'another cheat command',
    }]
  };
  it('should render component successfully', () => {
    const renderer = TestRenderer.create(<CheatCategory {...props} />);

    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
