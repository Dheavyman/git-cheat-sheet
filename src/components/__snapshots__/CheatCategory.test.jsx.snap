// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheatCategory component should render component successfully 1`] = `
<div
  className="category-card"
>
  <h6
    className="category"
  >
    cheat category
  </h6>
  <div
    className="cheat"
  >
    <p
      className="cheat-description"
    >
      cheat description
    </p>
    <p
      className="cheat-command"
    >
      $ 
      cheat command
    </p>
  </div>
  <div
    className="cheat"
  >
    <p
      className="cheat-description"
    >
      another cheat description
    </p>
    <p
      className="cheat-command"
    >
      $ 
      another cheat command
    </p>
  </div>
</div>
`;
